SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,430 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include "LoadBalancing/LayeredLBStrategy.h"
+#include "SpatialGDKTests/SpatialGDK/LoadBalancing/GridBasedLBStrategy/TestGridBasedLBStrategy.h"
+#include "TestLayeredLBStrategy.h"
+
+#include "Engine/Engine.h"
+#include "GameFramework/DefaultPawn.h"
+#include "GameFramework/GameStateBase.h"
+#include "Misc/Optional.h"
+#include "Tests/AutomationCommon.h"
+#include "Tests/AutomationEditorCommon.h"
+#include "Tests/TestDefinitions.h"
+
+#define LAYEREDLBSTRATEGY_TEST(TestName) \
+	GDK_TEST(Core, ULayeredLBStrategy, TestName)
+
+namespace
+{
+
+struct TestData {
+	ULayeredLBStrategy* Strat{ nullptr };
+	UWorld* TestWorld{ nullptr };
+	TMap<FName, AActor*> TestActors{};
+
+	~TestData()
+	{
+		ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(TestWorld->GetWorldSettings());
+		WorldSettings->WorkerLayers.Empty();
+	}
+};
+
+// Copied from AutomationCommon::GetAnyGameWorld().
+UWorld* GetAnyGameWorld()
+{
+	UWorld* World = nullptr;
+	const TIndirectArray<FWorldContext>& WorldContexts = GEngine->GetWorldContexts();
+	for (const FWorldContext& Context : WorldContexts)
+	{
+		if ((Context.WorldType == EWorldType::PIE || Context.WorldType == EWorldType::Game)
+			&& (Context.World() != nullptr))
+		{
+			World = Context.World();
+			break;
+		}
+	}
+
+	return World;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_ONE_PARAMETER(FWaitForWorld, TSharedPtr<TestData>, TestData);
+bool FWaitForWorld::Update()
+{
+	auto& TestWorld = TestData->TestWorld;
+	TestWorld = GetAnyGameWorld();
+
+	if (TestWorld && TestWorld->AreActorsInitialized())
+	{
+		AGameStateBase* GameState = TestWorld->GetGameState();
+		if (GameState && GameState->HasMatchStarted())
+		{
+			return true;
+		}
+	}
+
+	return false;
+}
+
+struct StrategyConfig {
+	TSubclassOf<UAbstractLBStrategy> StrategyClass;
+	TSet<TSoftClassPtr<AActor>> TargetActorTypes;
+};
```

Comment:
Wonder if it's worth having this struct since it's only used in one place to wrap two args as one?

Reply:
agreed, removed